### PR TITLE
Clean up try/catch/finally API

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Contributes to https://github.com/microsoft/typespec/issues/3825

Change CatchExpression to be CatchStatement since it is really a statement.
Make each of Try/Catch/Finally statements implement `IEnumerable<MethodBodyStatement>`. This is consistent with IfStatement/ForStatement,etc and it allows for writing generator code with structure that mirrors what the generated code will look like (apart from having comma delimiters as opposed to semicolons) e.g.

```c#
new TryStatement
{
 statement1,
 statement2
}
```
Generates

```c#
try
{
...
...
}
```

### CODE REVIEW

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/CatchStatement.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            using (writer.Scope())
+            {
+                foreach (var statement in Body)
+                {
+                    statement.Write(writer);
+                }
+            }
+        }
+
+        public CatchStatement Add(MethodBodyStatement statement)
```
Comment (m-nash) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/CatchStatement.cs:39]: I think we did this for consistency with If?  I am not sure I am fond of that pattern can we leave these off here and we can consider flipping the consistency the other way?

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/TryCatchFinallyStatement.cs [hunk 2]
```diff
@@ -9,46 +9,35 @@ namespace Microsoft.Generator.CSharp.Statements
...
+            Finally = @finally;
         }
 
-        public TryCatchFinallyStatement(MethodBodyStatement Try) : this(Try, Array.Empty<CatchExpression>(), null)
+        public TryCatchFinallyStatement(TryStatement @try) : this(@try, Array.Empty<CatchStatement>(), null)
         {
         }
 
-        public TryCatchFinallyStatement(MethodBodyStatement Try, CatchExpression Catch, MethodBodyStatement? Finally = null) : this(Try, [Catch], Finally)
+        public TryCatchFinallyStatement(TryStatement @try, CatchStatement catchStatement, FinallyStatement? @finally = null) : this(@try, new[] { catchStatement }, @finally)
```
Comment (m-nash) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/TryCatchFinallyStatement.cs:27]: name `tryStatement` and `finallyStatement` to match `catchStatement`?

**Review by m-nash — APPROVED** [review]

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/CatchStatement.cs (hunk 1 above)
Comment (m-nash) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/CatchStatement.cs:39]: nvm, I see the reason we need this

Comment (JoshLove-msft) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/CatchStatement.cs:39]: Needed to be able to use the list initializer.

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/TryCatchFinallyStatement.cs (hunk 2 above)
Comment (JoshLove-msft) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/TryCatchFinallyStatement.cs]: I meant to rename all to the shorter form.

**Pull request merged**